server/src/com/cloud/server/StatsCollector.java

```diff
@@ -285,9 +286,15 @@ private void init(Map<String, String> configs) {
                     s_logger.info(scheme + " is not a valid protocol for external statistics. No statistics will be send.");
                 }
 
-                externalStatsHost = uri.getHost();
+                if (!StringUtils.isEmpty(uri.getHost())) {
+                    externalStatsHost = uri.getHost();
+                }
+
                 externalStatsPort = uri.getPort();
-                externalStatsPrefix = uri.getPath().substring(1);
+
+                if (!StringUtils.isEmpty(uri.getPath())) {
+                    externalStatsPrefix = uri.getPath().substring(1);
+                }
 
                 /* Append a dot (.) to the prefix if it is set */
                 if (externalStatsPrefix != null && !externalStatsPrefix.equals("")) {
```

Comment:
I realize it's not in the scope of this PR, but would you consider changing this `if` block to use `StringUtils.isEmpty`?